### PR TITLE
Fix bad initial location of ckeditor format bar (BL-2514)

### PR DESCRIPTION
**Pull request by StephenMcConnel**
I think this also fixes BL-2515 and BL-2516, and a minor glitch in sometimes not bringing up the format bar when it should.

### CODE REVIEW

File: src/BloomBrowserUI/bookEdit/js/bloomEditing.js [hunk 1]
```diff
@@ -653,30 +653,56 @@ $(document).ready(function() {
...
+            // (Note that offsets are not defined if it's not visible.)
+            if (show) {
+                var barTop = bar.offset().top;
+                var boxTop = mapCkeditBoxTop[editor.id];
+                if (boxTop - barTop < 5) {
+                    var barLeft = bar.offset().left;
+                    var barHeight = bar.height();
+                    bar.offset({ top: boxTop - barHeight, left: barLeft });
+                }
+            }
```
Comment (JohnThomson) [inline src/BloomBrowserUI/bookEdit/js/bloomEditing.js]: Not sure exactly what the looked-up boxTop is relative to, but have you checked that this works if the scroll position changes between when the map is filled in and the ckEditor is shown? Also for page layouts (text-only page, which centers? Picture in middle, where bottom box is bottom-aligned?) where text editing changes the position of the text box...does that invalidate the saved map position?

Comment (StephenMcConnel) [inline src/BloomBrowserUI/bookEdit/js/bloomEditing.js]: I think the map is updated whenever the page is shown.  I've tried it on a variety of pages and page layouts.  I'm not sure how to bring about "text editing changes the position of the text box" however, or how to get the scroll position to change.  Our text boxes don't seem to scroll or change position no matter what I type, or how much I type.

Comment (JohnThomson) [inline src/BloomBrowserUI/bookEdit/js/bloomEditing.js]: I think if you make the window small enough you will start to get scroll bars on the whole page-edit area. That may be wrong; if you can't get them I suppose that isn't an issue.

Comment (JohnThomson) [inline src/BloomBrowserUI/bookEdit/js/bloomEditing.js]: It's probably unlikely enough that I think I will merge this. If you don't see any way to make scrolling happen you might add a note to the issue for testers to see if they can.

**Pull request merged**